californium-core/src/main/java/org/eclipse/californium/core/network/serialization/DataParser.java

```diff
@@ -162,8 +162,18 @@ public static void parseOptionsAndPayload(DatagramReader reader, Message message
 						Option option = new Option(currentOptionNumber);
 						option.setValue(reader.readBytes(optionLength));
 
-						// add option to message
-						message.getOptions().addOption(option);
+						if (currentOptionNumber == OptionNumberRegistry.CONTENT_FORMAT) {
+							// OptionSet.setContentFormat(int) API weird => cleanup on 3.0
```

Comment:
should be create an issue to not forget this ?

Reply:
Even if it get forgotten for the next couple of years, it will not harm too much :-).
But, yes, we can add a issue when merging this.